### PR TITLE
Split csi-driver-lvm binaries to two packages.

### PR DESCRIPTION
**Pull request by liulanze**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Split csi-driver-lvm binaries to two required packages, for the usage of creating golden container.

###### Change Log  

%changelog

Tue Sep 27 2022 Lanze Liu <lanzeliu@microsoft.com> - 0.4.1-2
- Split binaries into separate packages

Thu Sep 23 2022 Lanze Liu <lanzeliu@microsoft.com> - 0.4.1-1
- Original version for CBL-Mariner
- License Verified

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local build
- BuddyBuild

### CODE REVIEW

**Comment by liulanze** [comment]
BuddyBuild passed -> https://dev.azure.com/mariner-org/mariner/_build/results?buildId=243274&view=results

File: SPECS/csi-driver-lvm/csi-driver-lvm.spec [hunk 1]
```diff
@@ -20,9 +20,23 @@ Source0:        https://github.com/metal-stack/%{name}/archive/refs/tags/v%{vers
 #           -cf %%{name}-%%{version}-govendor.tar.gz vendor
 Source1:        %{name}-%{version}-govendor.tar.gz
 BuildRequires:  golang
+Requires:       %{name}-csi-lvmplugin-provisioner
```
Comment (Camelron) [inline SPECS/csi-driver-lvm/csi-driver-lvm.spec:23]: Curious, why split the package into two RPMS if they must both be installed together in order to function?

Comment (liulanze) [reply SPECS/csi-driver-lvm/csi-driver-lvm.spec:23]: For organization purposes, we will leverage these two packages to create docker images for golden container, so in the final, there will be two docker images pushed to the remote.

Comment (Camelron) [reply SPECS/csi-driver-lvm/csi-driver-lvm.spec:23]: Not sure I understand, will the two docker containers be shipped with these packages separately? My impression is that one will not function without the other, correct?

Comment (liulanze) [reply SPECS/csi-driver-lvm/csi-driver-lvm.spec:23]: Yes, that is right.

Comment (romoh) [reply SPECS/csi-driver-lvm/csi-driver-lvm.spec:23]: @Camelron, we are following a similar pattern in [cert-manager spec](https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/cert-manager/cert-manager.spec). Unless there's a difference guidance, it makes sense to keep components of the same opensource project that gets deployed together in the same spec.

**Review by Camelron — APPROVED** [review]
looks good to me

**Comment by liulanze** [comment]
Thanks, I noticed I don't have write access to merge the PR, please help to merge it, thanks.

**Comment by mandeepsplaha** [comment]
Merging this PR to main on Lanze's behalf since he doesn't have the write access yet.

**Pull request merged**